pyre/extra_tests/parity_tests/mapdict_devolved_raising_eq.py

```
@@ -0,0 +1,137 @@
+    except ValueError:
+        pass
+    else:
+        raise AssertionError("raising __eq__ swallowed on an alternate read path")
+
+# The dict subscript itself agrees.
+try:
+    r.__dict__["zz"]
+except ValueError:
+    pass
+except KeyError:
+    raise AssertionError("raising __eq__ reported as a missing key")
```

Ruff (0.16.1)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)